bundles/org.openhab.binding.luxtronikheatpump/src/main/java/org/openhab/binding/luxtronikheatpump/internal/LuxtronikHeatpumpHandlerFactory.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.luxtronikheatpump.internal;
+
+import static org.openhab.binding.luxtronikheatpump.internal.LuxtronikHeatpumpBindingConstants.*;
+
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.i18n.LocaleProvider;
+import org.openhab.core.i18n.TranslationProvider;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseThingHandlerFactory;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerFactory;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * The {@link LuxtronikHeatpumpHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Stefan Giehl - Initial contribution
+ */
+@NonNullByDefault
+@Component(configurationPid = "binding.luxtronikheatpump", service = ThingHandlerFactory.class)
+public class LuxtronikHeatpumpHandlerFactory extends BaseThingHandlerFactory {
+
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Set.of(THING_TYPE_HEATPUMP);
+    private final LuxtronikTranslationProvider translationProvider;
+
+    @Activate
+    public LuxtronikHeatpumpHandlerFactory(final @Reference LocaleProvider localeProvider,
+            final @Reference TranslationProvider i18nProvider, ComponentContext componentContext) {
+        super.activate(componentContext);
```

Comment:
Are you sure this call is necessary?

Reply:
guess that was a copy/paste from some example. will check if it can be removed...

Reply:
seems removing it results in this error when the binding is loaded:
```
2021-03-21 17:53:27.239 [ERROR] [rg.openhab.binding.luxtronikheatpump] - bundle org.openhab.binding.luxtronikheatpump:3.1.0.202103211652 (209)[org.openhab.binding.luxtronikheatpump.internal.LuxtronikHeatpumpHandlerFactory(361)] :  Error during instantiation of the implementation object
java.lang.reflect.InvocationTargetException: null
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]
	at org.apache.felix.scr.impl.inject.ComponentConstructor.newInstance(ComponentConstructor.java:309) ~[bundleFile:?]
...

```

Reply:
That's strange. This code works for me:
```    @Activate
    public LuxtronikHeatpumpHandlerFactory(final @Reference LocaleProvider localeProvider,
            final @Reference TranslationProvider i18nProvider) {
        this.translationProvider = new LuxtronikTranslationProvider(getBundleContext().getBundle(), i18nProvider,
                localeProvider);
    }
```

Reply:
the binding compiles for me as well without that line. But when replacing the built binding on my test server it throws the exception above when the new file is loaded. That doesn't happen when the line is present :thinking: